flink-tests/src/test/java/org/apache/flink/test/recovery/JobManagerHAProcessFailureRecoveryITCase.java

```diff
@@ -220,6 +221,187 @@ public void flatMap(Long value, Collector<Long> out) throws Exception {
 		env.execute();
 	}
 
+	/**
+	 * Tests that the JobManager logs failures during recovery properly.
+	 *
+	 * @see <a href="https://issues.apache.org/jira/browse/FLINK-3185">FLINK-3185</a>
+	 */
+	@Test
+	public void testJobManagerRecoveryFailureLog() throws Exception {
```

Comment:
I think it would be better to test the behaviour more fine grained. What we actually tested was that the recovery logs a message if there is a recovery failure due to non-existing state handles. However, the behaviour slightly changed such that a failing recovery will lead to a `Dispatcher` termination (see `DispatcherHATest#testFailingRecoveryIsAFatalError`). Thus, I think this test is no longer necessary and can be removed.